pkg/streamingpromql/operators/functions/absent_over_time.go

```diff
@@ -0,0 +1,124 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+// Provenance-includes-location: https://github.com/prometheus/prometheus/blob/main/promql/functions.go
+// Provenance-includes-license: Apache-2.0
+// Provenance-includes-copyright: The Prometheus Authors.
+
+package functions
+
+import (
+	"context"
+
+	"github.com/prometheus/prometheus/promql"
+	"github.com/prometheus/prometheus/promql/parser"
+	"github.com/prometheus/prometheus/promql/parser/posrange"
+
+	"github.com/grafana/mimir/pkg/streamingpromql/limiting"
+	"github.com/grafana/mimir/pkg/streamingpromql/types"
+)
+
+// AbsentOverTime performs a rate calculation over a range vector.
+type AbsentOverTime struct {
+	timeRange                types.QueryTimeRange
+	argExpressions           parser.Expr
+	inner                    types.RangeVectorOperator
+	expressionPosition       posrange.PositionRange
+	memoryConsumptionTracker *limiting.MemoryConsumptionTracker
+	presence                 []bool
+	exhausted                bool
```

Comment:
[nit] We usually export some of these - could you please make this consistent with the other operators? The usual rule of thumb is that internal implementation details aren't exported.

Reply:
Updated in 4726cddd0f329867b1d4b3b1acc803be52e8fb17